Source/src/WixSharp/Nsis/WinVer/OSValidation.cs

```diff
@@ -0,0 +1,142 @@
+        /// </summary>
+        public bool TerminateInstallation { get; set; } = true;
+
+        /// <summary>
+        /// Minimal required windows version
+        /// </summary>
+        // ReSharper disable once MemberCanBePrivate.Global
+        // ReSharper disable once UnusedAutoPropertyAccessor.Global
+        public WindowsVersionNumber? MinVersion { get; set; }
+
+        /// <summary>
+        /// Optional windows versions that are not supported
+        /// </summary>
+        // ReSharper disable once MemberCanBePrivate.Global
+        // ReSharper disable once CollectionNeverUpdated.Global
+        public IList<WindowsVersion> UnsupportedVersions { get; } = new List<WindowsVersion>();
+
+        internal bool Any => MinVersion.HasValue || UnsupportedVersions.Any();
+
+        internal string BuildVersionCheckScriptPart()
+        {
+            if (!Any)
+            {
+                return string.Empty;
+            }
+
+            var sb = new StringBuilder();
+            using (var writer = new StringWriter(sb))
+            {
+                ConstructScript(writer);
+                return writer.ToString();
+            }
+        }
+
+        private void ConstructScript(StringWriter writer)
+        {
+            WriteMinVersion(writer);
+
+            WriteUnsupportedVersions(writer);
+
+            WriteMessageBox(writer);
+
+            if (TerminateInstallation)
+            {
+                writer.WriteLine("goto end");
+            }
+
+            writer.WriteLine(MinVersion.HasValue ? "${EndUnless}" : "${EndIf}");
+        }
+
+        private void WriteMinVersion(StringWriter writer)
+        {
+            if (!MinVersion.HasValue)
+            {
+                writer.Write("${If} ");
+                return;
+            }
+
+            writer.Write("${Unless} ${AtLeastWin");
+            var minVersion = MinVersion.Value;
+            writer.WriteLine($"{minVersion.GetDescription()}}}");
+
+            if (UnsupportedVersions.Any())
+            {
+                writer.Write("${OrIf} ");
+            }
+        }
+
+        private void WriteUnsupportedVersions(StringWriter writer)
+        {
+            for (var i = 0; i < UnsupportedVersions.Count; i++)
+            {
+                if (i > 0)
+                {
+                    writer.Write("${OrIf} ");
+                }
+
+                var unsupportedVersion = UnsupportedVersions[i];
+
+                writer.Write("${IsWin");
+
+                string winVersion = unsupportedVersion.VersionNumber.GetDescription();
+                writer.WriteLine($"{winVersion}}}");
+
+                if (unsupportedVersion.ServicePack != -1)
+                {
+                    writer.WriteLine($"${{AndIf}} ${{IsServicePack}} {unsupportedVersion.ServicePack}");
+                }
+            }
+        }
+
+        private void WriteMessageBox(StringWriter writer)
+        {
```

Comment:
I'd suggest to check the command line parameters in the runtime, and in case "/S" switch is present skip the message box creation

Reply:
Sure, thanks, forgot about it.

Reply:
Done